### PR TITLE
DBZ-4502 Check GTID changes when current GTID is undefined

### PR DESCRIPTION
**Pull request by kgalieva**
An infinite while loop can happen on connector restart when the first event after the restart is a transaction commit event and the current GTID is still undefined.
The solution is to check that the GTID changes when the current GTID is undefined.

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotContext.java [hunk 1]
```diff
@@ -145,4 +153,8 @@ public ExecuteSnapshotKafkaSignal getExecuteSnapshotSignals() {
     public boolean hasExecuteSnapshotSignals() {
         return !executeSnapshotSignals.isEmpty();
     }
+
+    public boolean watermarksChanged() {
+        return !previousLowWatermark.equals(lowWatermark) || !previousHighWatermark.equals(highWatermark);
```
Comment (jpechane) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotContext.java:158]: Is it guaranteed that previous watermarks will be always not null?

Comment (kgalieva) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotContext.java:158]: Yes, this watermarks change check is done after the `closeWindow()` call, so previous watermarks will always be initialized

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotContext.java:158]: ok, thanks

**Comment by jpechane** [comment]
@kgalieva Applied, thanks a lot!

**Pull request merged**